core/runtime/src/main/java/io/quarkus/runtime/configuration/FileSystemOnlySourcesConfigBuilder.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.runtime.configuration;
+
+import static io.smallrye.config.PropertiesConfigSourceLoader.inFileSystem;
+
+import java.nio.file.Paths;
+
+import io.smallrye.config.SmallRyeConfigBuilder;
+
+public class FileSystemOnlySourcesConfigBuilder implements ConfigBuilder {
+    @Override
+    public SmallRyeConfigBuilder configBuilder(final SmallRyeConfigBuilder builder) {
+        return builder.withSources(
```

Comment:
You want to disable the default sources which include the classpath lookup and then add the `inFileSystem`: `builder.setAddDefaultSources(false).addSystemSources().withSources(inFileSystem...)`